kratos/utilities/condition_number_utility.h

```diff
@@ -84,27 +87,44 @@ class ConditionNumberUtility
     ///@{
 
     /// Default constructor.
-    ConditionNumberUtility() 
-    {
-        
-    }
+    ConditionNumberUtility()
+        :mpEigenSolverMax(nullptr), 
+         mpEigenSolverMin(nullptr)
+    {}
+    
+    /// Default constructor.
+    ConditionNumberUtility(
+        LinearSolverType::Pointer pEigenSolverMax,
+        LinearSolverType::Pointer pEigenSolverMin
+        ) :mpEigenSolverMax(pEigenSolverMax), 
+           mpEigenSolverMin(pEigenSolverMin)
+    {}
 
     /// Destructor
-    ~ConditionNumberUtility()
-    {
-
-    }
+    virtual ~ConditionNumberUtility(){}
 
     ///@}
     ///@name Operations
     ///@{
     
     /**
      * This function computes using the inverse power method the minimal eigenvalue
-     * @param InputMatrix: The matrix to compute the eigenvalue
-     * @param pEigenSolverMax: The solver to get the maximal eigen value
-     * @param pEigenSolverMin: The solver to get the minimal eigen value
-     * @return condition_number: The condition number
+     * @param InputMatrix The matrix to compute the eigenvalue
+     * @return condition_number The condition number
+     */
+    
+    double GetConditionNumber(SparseMatrixType& InputMatrix)
+    {
+        KRATOS_ERROR_IF(mpEigenSolverMax == nullptr || mpEigenSolverMin == nullptr) << "WARNING:: PLEASE DEFINE THE EigenSolvers" << std::endl;
```

Comment:
If is an error, I would remove the "WARNING"

Reply:
OK